BuildAddOn.py

```diff
@@ -45,33 +50,42 @@ def PrepareParameters (args):
     configData = json.load (configFile)
     addOnName = configData['addOnName']
     acVersionList = None
-    languageList = None
 
     if args.acVersion:
         acVersionList = args.acVersion
     else:
         acVersionList = []
-        for version in configData['devKitLinks'][platformName]:
+        for version in devKitData[platformName]:
             acVersionList.append (version)
 
     # Get needed language codes
-    if args.release:
-        configLangUpper = [lang.upper () for lang in configData['languages']]
-        languageList = ['ALL']
-        if args.language is not None:
-            languageList = [lang.upper () for lang in args.language]
-
-        if 'ALL' in languageList:
-            languageList = configLangUpper
-        else:
-            for lang in languageList:
-                if lang not in configLangUpper:
-                    raise Exception ('Language not supported!')
+    languageList = [].append (configData['defaultLanguage'])
+    if args.allLocalizedVersions:
+        languageList = [lang.upper () for lang in configData['languages']]
+                
+    # Get additional CMake parameters
+    additionalParams = None
+    if 'additionalCMakeParams' in configData or args.additionalCMakeParams:
+        additionalParams = {}
+
+        if 'additionalCMakeParams' in configData:
+            for key in configData['additionalCMakeParams']:
+                additionalParams[key] = configData['additionalCMakeParams'][key]
+
+        if args.additionalCMakeParams:
+            for param in args.additionalCMakeParams:
+                if '=' not in param:

```

Comment:
I thinks that's what I originally thought about "python-like" arg usage. I guess this could work like the _allLocalizedVersions_ arg **too**, so they just set a custom parameter without a "=value" pair, and we pass it with the value _ON_ to the cmake project generation. Obviously, we want to support the "=" cases too

Reply:
Added option for flag-like additional arguments